DownFall/Extensions/CGPoint+Extension.swift

```diff
@@ -111,25 +126,60 @@ extension CGPoint {
         let y: CGFloat
         switch verticaliy {
         case .top:
-            y = that.center.y + that.height/2 + this.height/2 + yOffset
+            y = that.maxY + this.height/2 + yOffset
         case .center:
             y = that.center.y
         case .bottom:
-            y = that.center.y - that.height - this.height/2 - yOffset
+            y = that.minY - this.height/2 - yOffset
         }
         
         let x: CGFloat
         switch anchor {
         case .left:
-            x = that.center.x - that.width/2 - this.width/2 - xOffset
+            x = that.minX - this.width/2 - xOffset
         case .right:
-            x = that.center.x + that.width/2 + this.width/2 + xOffset
+            x = that.maxX + this.width/2 + xOffset
+        case .center:
+            x = that.center.x
         }
         
         return CGPoint(x: x, y: y)
     }
     
     
+    static func position(_ this: CGRect,
+                         inside that: CGRect,
+                         verticaliy: Verticality,
+                         anchor: Anchor,
+                         xOffset: CGFloat = 0.0,
+                         yOffset: CGFloat = 0.0) -> CGPoint {
+        
+        
+        let y: CGFloat
+        switch verticaliy {
+        case .top:
+            y = that.height/2 - this.height/2 - yOffset
+        case .center:
+            y = 0.0
+        case .bottom:
+            y = -that.height/2 + this.height/2 + yOffset
+        }
+        
+        let x: CGFloat
+        switch anchor {
+        case .left:
+            x = -that.width/2 + this.width/2 + xOffset
+        case .center:
+            x = 0.0
+        case .right:
+            x = that.width/2 - this.width/2 - xOffset
+        }
+        
+        return CGPoint(x: x, y: y)
+    }
```

Comment:
This needs to be translated to the parents coordinate system or else it doesn't always work correctly

Reply:
Basically there are two types of methods we need to support.

1) You, the parent, need to place a child in your coord system. This is a simple case where we don't need to do any translating
2) You, the parent, need to place a child relative to another child.  This is slightly more complex.  You need to account for the center of `that` child because `that` child already lives in the parent's coord system.